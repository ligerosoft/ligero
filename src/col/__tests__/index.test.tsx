import { mount } from 'enzyme';
import React from 'react';
import Col from '..';
import Row from '../../row';

describe('Col', () => {
  it('mount col correctly', () => {
    expect(mount(<Col />)).toMatchSnapshot();
  });

  it('render col with span', () => {
    const wrapper = mount(
      <Row>
        <Col span={8}>col 8</Col>
        <Col span={8}>col 8</Col>
        <Col span={8}>col 8</Col>
      </Row>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('render gutter correctly', () => {
    const wrapper = mount(
      <Row gutter={20}>
        <Col span={6}>6</Col>
        <Col span={12}>12</Col>
        <Col span={6}>6</Col>

        <Col span={8}>8</Col>
        <Col span={8}>8</Col>
        <Col span={8}>8</Col>

        <Col span={4} offset={4}>
          8
        </Col>
        <Col span={6}>8</Col>
        <Col span={10}>8</Col>
      </Row>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
