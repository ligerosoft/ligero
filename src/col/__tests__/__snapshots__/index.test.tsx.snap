// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col mount col correctly 1`] = `
<@ligero/col>
  <div
    className="ligero-col"
    style={Object {}}
  />
</@ligero/col>
`;

exports[`Col render col with span 1`] = `
<@ligero/row>
  <div
    className="ligero-row ligero-row--wrap"
  >
    <@ligero/col
      index={0}
      key="0"
      span={8}
    >
      <div
        className="ligero-col ligero-col--8"
        style={Object {}}
      >
        col 8
      </div>
    </@ligero/col>
    <@ligero/col
      index={1}
      key="1"
      span={8}
    >
      <div
        className="ligero-col ligero-col--8"
        style={Object {}}
      >
        col 8
      </div>
    </@ligero/col>
    <@ligero/col
      index={2}
      key="2"
      span={8}
    >
      <div
        className="ligero-col ligero-col--8"
        style={Object {}}
      >
        col 8
      </div>
    </@ligero/col>
  </div>
</@ligero/row>
`;

exports[`Col render gutter correctly 1`] = `
<@ligero/row
  gutter={20}
>
  <div
    className="ligero-row ligero-row--wrap"
  >
    <@ligero/col
      index={0}
      key="0"
      span={6}
    >
      <div
        className="ligero-col ligero-col--6"
        style={
          Object {
            "paddingLeft": undefined,
            "paddingRight": 10,
          }
        }
      >
        6
      </div>
    </@ligero/col>
    <@ligero/col
      index={1}
      key="1"
      span={12}
    >
      <div
        className="ligero-col ligero-col--12"
        style={
          Object {
            "paddingLeft": 10,
            "paddingRight": 0,
          }
        }
      >
        12
      </div>
    </@ligero/col>
    <@ligero/col
      index={2}
      key="2"
      span={6}
    >
      <div
        className="ligero-col ligero-col--6"
        style={
          Object {
            "paddingLeft": 20,
            "paddingRight": -10,
          }
        }
      >
        6
      </div>
    </@ligero/col>
    <@ligero/col
      index={3}
      key="3"
      span={8}
    >
      <div
        className="ligero-col ligero-col--8"
        style={
          Object {
            "paddingLeft": undefined,
            "paddingRight": 10,
          }
        }
      >
        8
      </div>
    </@ligero/col>
    <@ligero/col
      index={4}
      key="4"
      span={8}
    >
      <div
        className="ligero-col ligero-col--8"
        style={
          Object {
            "paddingLeft": 10,
            "paddingRight": 0,
          }
        }
      >
        8
      </div>
    </@ligero/col>
    <@ligero/col
      index={5}
      key="5"
      span={8}
    >
      <div
        className="ligero-col ligero-col--8"
        style={
          Object {
            "paddingLeft": 20,
            "paddingRight": -10,
          }
        }
      >
        8
      </div>
    </@ligero/col>
    <@ligero/col
      index={6}
      key="6"
      offset={4}
      span={4}
    >
      <div
        className="ligero-col ligero-col--4 ligero-col--offset-4"
        style={
          Object {
            "paddingLeft": undefined,
            "paddingRight": 10,
          }
        }
      >
        8
      </div>
    </@ligero/col>
    <@ligero/col
      index={7}
      key="7"
      span={6}
    >
      <div
        className="ligero-col ligero-col--6"
        style={
          Object {
            "paddingLeft": 10,
            "paddingRight": 0,
          }
        }
      >
        8
      </div>
    </@ligero/col>
    <@ligero/col
      index={8}
      key="8"
      span={10}
    >
      <div
        className="ligero-col ligero-col--10"
        style={
          Object {
            "paddingLeft": 20,
            "paddingRight": -10,
          }
        }
      >
        8
      </div>
    </@ligero/col>
  </div>
</@ligero/row>
`;
